frontend/src/components/create-user-page/CreateUser.tsx

```diff
@@ -61,49 +98,81 @@ export const CreateUser: React.FunctionComponent = () => {
             `${loginContext.username}:${loginContext.password}`
           )}`,
         },
-        body: JSON.stringify(FormData),
-      }).then((res) => res.json());
+        body: JSON.stringify({
+          name: formName,
+          username: formUsername,
+          email: formEmail,
+          password: formPassword,
+        }),
+      })
+        .then((res) => {
+          setAddUserMessage(
+            res.status == 200
+              ? "User added successfully!"
+              : "Sorry, Couldn't add the user!"
+          );
+        })
+        .then(() => (e.target as HTMLFormElement).reset());
     }
   };
 
   return (
-    <div>
+    <div className="form">
       <form onSubmit={handleSubmit}>
-        <label>
-          Name:
-          <input
-            type="text"
-            name="Name"
-            onChange={(e) => setFormName(e.target.value)}
-          />
-        </label>
-        <label>
-          Username:
-          <input
-            type="text"
-            name="Username"
-            onChange={(e) => setFormUsername(e.target.value)}
-          />
-        </label>
-        <label>
-          Email:
-          <input
-            type="email"
-            name="Email"
-            onChange={(e) => setFormEmail(e.target.value)}
-          />
-        </label>
-        <label>
-          Password:
-          <input
-            type="password"
-            name="Password"
-            onChange={(e) => setFormPassword(e.target.value)}
-            onBlur={(e) => validatePassword(e.target.value)}
-          />
-        </label>
-        <input type="submit" value="Create User" />
+        <h1 className="form_heading">Create New User</h1>
+        <div className="form_field">
+          <div className="form_field--row">
+            <label className="form_field--label">Name:</label>
+            <input
+              className="form_field--input"
+              type="text"
+              name="Name"
+              onChange={(e) => validateName(e.target.value)}
+            />
+          </div>
+          <p className="message">{nameMessage}</p>
```

Comment:
All of these elements should only render if the corresponding message is non-empty, otherwise you'll have a permanent empty `<p>` on the page, confusing the DOM
```suggestion
          {nameMessage ? <p className="message">{nameMessage}</p> : <></>}
```

Reply:
Apply the above to the rest of the file too please!